### PR TITLE
Allowing more control in offboard mode

### PR DESCRIPTION
**Pull request by AlexisTM**
This PR enable offboard control with the following data (x,y,vz) and (vx,vy,z) following the issue https://github.com/PX4/Firmware/issues/8117

### CODE REVIEW

**Comment by dagar** [comment]
This looks good to me, but I'm not a heavy offboard user.
Would you be able to add some simple testing to run through everything in offboard? 

Testing options
 - this needs a cleanup, but there's existing ROS based testing here. https://github.com/PX4/Firmware/tree/master/integrationtests
 - also looking at using Dronecore (https://github.com/dronecore/dronecore).

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -4028,17 +4032,18 @@ set_control_mode()
 		  !status.in_transition_mode;
 
 		control_mode.flag_control_velocity_enabled = (!offboard_control_mode.ignore_velocity ||
-			!offboard_control_mode.ignore_position) && !status.in_transition_mode &&
+			!offboard_control_mode.ignore_altitude) && !status.in_transition_mode &&
```
Comment (AlexisTM) [inline src/modules/commander/commander.cpp]: This should stay `!offboard_control_mode.ignore_position` !

**Comment by AlexisTM** [comment]
I totally agree. The offboard mode is not tested for every case. The offboard mode also requires some rework; I personally do not like the messages needed. 

I would do 3 type of setpoints as the 3 are used in different conditions and it is very unlikely that you switch from one to another.

- Acceleration SP: I never used it, yet could be used
- Position/Velocity SP: Velocities for Takeoff, land, obstacle avoidance and position to hold the position
- Attitude SP: Some [ROS controllers](https://github.com/ethz-asl/mav_control_rw) features a (complex) trajectory following and outputs attitude setpoints. 

Another idea would be to send position triplets to be able to use the same control_auto() function which takes in account what will happen at the setpoint. (to avoid overshoot and do better angles).

**Comment by dagar** [comment]
Have you seen this? https://github.com/mavlink/mavlink/pull/670

**Pull request closed without merging**